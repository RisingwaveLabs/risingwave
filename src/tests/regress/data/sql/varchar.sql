--
-- VARCHAR
--

CREATE TABLE VARCHAR_TBL_TMP(f1 varchar);

INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('a');

INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('A');

-- any of the following three input formats are acceptable
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('1');

INSERT INTO VARCHAR_TBL_TMP (f1) VALUES (2);

INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('3');

-- zero-length char
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('');

-- try varchar's of greater than 1 length
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('cd');
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('c     ');


SELECT * FROM VARCHAR_TBL_TMP;

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 <> 'a';

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 = 'a';

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 < 'a';

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 <= 'a';

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 > 'a';

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 >= 'a';

DROP TABLE VARCHAR_TBL_TMP;

--
-- Now test longer arrays of char
--
-- This varchar_tbl was already created and filled in test_setup.sql.
--

SELECT * FROM VARCHAR_TBL;
