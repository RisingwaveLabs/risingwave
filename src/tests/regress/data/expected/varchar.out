--
-- VARCHAR
--
CREATE TABLE VARCHAR_TBL_TMP(f1 varchar);
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('a');
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('A');
-- any of the following three input formats are acceptable
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('1');
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES (2);
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('3');
-- zero-length char
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('');
-- try varchar's of greater than 1 length
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('cd');
INSERT INTO VARCHAR_TBL_TMP (f1) VALUES ('c     ');
SELECT * FROM VARCHAR_TBL_TMP;
   f1   
--------
 a
 A
 1
 2
 3
 
 cd
 c     
(8 rows)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 <> 'a';
   f1   
--------
 A
 1
 2
 3
 
 cd
 c     
(7 rows)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 = 'a';
 f1 
----
 a
(1 row)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 < 'a';
 f1 
----
 A
 1
 2
 3
 
(5 rows)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 <= 'a';
 f1 
----
 a
 A
 1
 2
 3
 
(6 rows)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 > 'a';
   f1   
--------
 cd
 c     
(2 rows)

SELECT c.*
   FROM VARCHAR_TBL_TMP c
   WHERE c.f1 >= 'a';
   f1   
--------
 a
 cd
 c     
(3 rows)

DROP TABLE VARCHAR_TBL_TMP;
--
-- Now test longer arrays of char
--
-- This varchar_tbl was already created and filled in test_setup.sql.
--
SELECT * FROM VARCHAR_TBL;
    f1    
----------
 a
 ab
 abcd
 abcd    
(4 rows)

